3rdparty/everest/library/Hacl_Curve25519_joined.c

```diff
@@ -24,6 +24,7 @@
 #else
 #include MBEDTLS_CONFIG_FILE
 #endif
+#include "mbedtls/check_config.h"
```

Comment:
[As I've stated before](https://github.com/ARMmbed/mbedtls/pull/1999#issuecomment-615460281), I don't like this approach. Every library source file is changed to include something that doesn't really concern it.

Furthermore this is unrelated to the original objective of the pull request. It neither fixes the problem nor contributes to making the problem less likely to happen again.

I wouldn't have vetoed the PR on this basis, but since the PR needs rework anyway, please remove the inclusion of `check_config.h` everywhere.

To include `check_config.h` everywhere, on `development`, add it to `library/common.h`, which [was added to `development` today](https://github.com/ARMmbed/mbedtls/pull/3121) (and we'll also need to include `common.h` everywhere, which isn't done yet).